.github/workflows/ci.yaml

```diff
@@ -0,0 +1,150 @@
+          repository: apache/arrow
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+      - name: Setup Archery
+        run: |
+          pip install -e arrow/dev/archery[lint]
+      - name: Run RAT
+        run: |
+          cd datafusion-ruby && archery lint --rat
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "lts/*"
+      - name: Prettier
+        run: |
+          npx prettier --check "**/*.md"
+
+  test:
+    name: Test
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version:
+          - '3.1'
+        runs-on:
+          - ubuntu-latest
+          - macos-latest
+          # - windows-latest
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: actions/checkout@v3
+
+      # Install datafusion-c-glib
+      # We can use packages once datafusion-c-glib is released.
+      - uses: actions/checkout@v3
+        with:
+          repository: 'datafusion-contrib/datafusion-c'
+          path: datafusion-c
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Prepare on Ubuntu
+        if: |
+          matrix.runs-on == 'ubuntu-latest'
+        run: |
+          sudo apt update
+          sudo apt install -y -V \
+            ca-certificates \
+            lsb-release \
+            wget
+          wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+          sudo apt install -y -V \
+            ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+          sudo apt update
+          sudo apt install -y -V \
+            libarrow-glib-dev \
+            libgirepository1.0-dev \
+            ninja-build
+          pip install meson
+      - name: Prepare on macOS
+        if: |
+          matrix.runs-on == 'macos-latest'
+        run: |
+          brew update --preinstall
+          brew bundle --file=datafusion-c/Brewfile
+      - uses: actions-rs/toolchain@v1
+        id: rust-toolchain
+        with:
+          override: true
+          toolchain: stable
+      - name: Cache Cargo
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo
+          key: cargo-test-${{ matrix.runs-on }}-${{ steps.rust-toolchain.outputs.rustc_hash }}-${{ hashFiles('datafusion-c/**/Cargo.toml') }}
+      - name: Install cargo-c
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-c
+      - name: Setup
+        run: |
+          meson setup \
+            --libdir=lib \
+            --prefix=$PWD/local \
+            datafusion-c-build \
+            datafusion-c
+      - name: Build
+        run: |
+          ninja -C datafusion-c-build
+      - name: Install
+        run: |
+          ninja -C datafusion-c-build install
```

Comment:
[random] sorry, may I ask after publish as a gem, how `bundle install` compile the `datafusion-c`? I imagine there should be a `ext` folder containing the C code inside the gem?

Reply:
`bundle install` doesn't compile datafusion-c. Users need to install datafusion-c separately. But I'll add auto system package (`libdatafusion-glib-dev`) install feature to gem like `red-parquet` does:
* https://github.com/apache/arrow/blob/master/ruby/red-parquet/red-parquet.gemspec#L42
* https://github.com/apache/arrow/blob/master/ruby/red-parquet/dependency-check/Rakefile

(I don't bundle datafusion-c to this implementation.)

FYI: We don't need C code in this implementation because this implementation uses GObject Introspection to generate bindings. datafusion-glib https://github.com/datafusion-contrib/datafusion-c/tree/main/datafusion-glib ([GLib](https://docs.gtk.org/glib/) integration layer) in datafusion-c provides metadata to generate bindings dynamically. This implementation uses the metadata to generate bindings dynamically at run-time.

See also: Ruby bindings 2016 - How to create bindings 2016: https://slide.rabbit-shocker.org/authors/kou/rubykaigi-2016/